encryption.md

```diff
@@ -0,0 +1,52 @@
+## Message Encryption
+
+DIDComm supports two types of message encryption: Authenticated Sender Encryption and Anonymous Sender Encryption. Both forms are encrypted to the recipient, but only Authenticated Sender Encryption provides assurances of who the sender is. 
+
+The encrypted form of a JWM is a JWE. The JOSE family defines [JSON Web Algorithms](https://tools.ietf.org/html/rfc7518) (JWAs) which standardize certain cryptographic operations that are related to preparing JOSE structures. For the purposes of interoperability, DIDComm messaging does not support all JWAs, rather it takes a subset of the supported algorithms that are applicable for the following cases around secure messaging. These supported algorithms are listed here.
+
+
+### Sender Authenticated Encryption
+
+When a sender would like to encrypt a message for a recipient or multiple recipients and also be authenticated such that the recipients can determine they were the party that encrypted the message, the JWA of `ECDH-1PU` defined by [draft](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-02) SHOULD be used within the structure of a JWE. 
+
+For the keys involved in key agreement, the following elliptic curves MUST be supported.
+
+| Curve  | Description                                                  |
+| ------ | ------------------------------------------------------------ |
+| X25519 | The underlying curve is actually `Curve25519`, however when used in the context of Diffie-Hellman the identifier of `X25519` is used |
+| P-256  | NIST defined P-256 elliptic curve                            |
+
+For content encryption of the message, the following algorithms MUST be supported.
+
+| Algorithm(JWA) | Description                                           |
+| -------------- | ----------------------------------------------------- |
+| XC20P          | XChaCha20Poly1305                                     |
+| AES-GCM        | Advanced Encryption Standard with galois/counter mode |
+
+TODO: Include language about safe nonce considerations.
+
+### Anonymous Encryption
+
+When a sender would like to encrypt a message for a recipient or multiple recipients but not be authenticated by the recipients as the party who encrypted the message, the JWA of `ECDH-ES` defined by [RFC 7518](https://tools.ietf.org/html/rfc7518#section-4.6) SHOULD be used within the structure of a JWE. 
+
+For the keys involved in key agreement, the following elliptic curves MUST be supported.
+
+| Curve  | Description                                                  |
+| ------ | ------------------------------------------------------------ |
+| X25519 | The underlying curve is actually `Curve25519`, however when used in the context of Diffie-Hellman the identifier of `X25519` is used |
+| P-256  | NIST defined P-256 elliptic curve                            |
```

Comment:
Although `P-256` curve is the correct name, JWA also specifies the key wrapping algorithm using this curve. It's better to mention the algorithm value, especially that we're talking about ECDH here.

For instance, Anonymous encryption would be using ECDH-ES where the sender info is not required and a new ephemeral key is created at the send side for each new encryption. The ECDH key wrapping algorithm is called: `ECDH-ES+A256KW` (as per the [JWA spec](https://tools.ietf.org/html/rfc7518#section-4.6)).

As the key wrapping comment for ECDH-1PU above, it would help if we have a key wrapping algorithms table.

Reply:
+1 for an algorithms table, cross linking: https://github.com/w3c-ccg/lds-jws2020/issues/12

We want to list the JsonWebKey recommended encryption `alg` here as well, and we need to ensure we support all the types that DIDComm uses.

Reply:
What form does this table show up in? I'm all for adding clarity here, but I have no idea how it ought to be properly represented in this document.
If you can help guide me, I'd like to get this done. I'm also open to accepting this PR as is and making the clarifications in another PR.
@Baha-sk @OR13

Reply:
I guess a new key wrapping algorithms table would look like this :
```
| KW Algorithm    | Curve (epk crv) | key type (epk kty) | Description                                                  |
| --------------- | --------------- | ------------------ | ------------------------------------------------------------ |
| ECDH-ES+A256KW  | P-256  | EC  | ECDH-ES key wrapping using key with NIST defined P-256 elliptic curve to create a 256 bits key as defined in [7518](https://tools.ietf.org/html/rfc7518#section-4.6.2)       |
| ECDH-ES+A256KW  | P-384  | EC  | ECDH-ES key wrapping using key with NIST defined P-384 elliptic curve to create a 256 bits key as defined in [7518](https://tools.ietf.org/html/rfc7518#section-4.6.2)       |
| ECDH-ES+A256KW  | P-521  | EC  | ECDH-ES key wrapping using key with NIST defined P-521 elliptic curve to create a 256 bits key as defined in [7518](https://tools.ietf.org/html/rfc7518#section-4.6.2)       |
| ECDH-ES+A256KW  | X25519 | OKP | ECDH-ES with X25519 ([RFC7748 section 5](https://tools.ietf.org/html/rfc7748#section-5)) to create a 256 bits key. The underlying curve is actually `Curve25519`, however when used in the context of Diffie-Hellman the identifier of `X25519` is used |
| ECDH-1PU+A256KW | P-256  | EC  | ECDH-1PU key wrapping using key with NIST defined P-256 elliptic curve to create a 256 bits key as defined in [ecdh-1pu](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-03#section-2.2)      |
| ECDH-1PU+A256KW | P-384  | EC  | ECDH-1PU key wrapping using key with NIST defined P-384 elliptic curve to create a 256 bits key as defined in [ecdh-1pu](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-03#section-2.2)      |
| ECDH-1PU+A256KW | P-521  | EC  | ECDH-1PU key wrapping using key with NIST defined P-521 elliptic curve to create a 256 bits key as defined in [ecdh-1pu](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-03#section-2.2)      |
| ECDH-1PU+A256KW | X25519 | OKP | ECDH-1PU X25519 ([RFC7748 section 5](https://tools.ietf.org/html/rfc7748#section-5)) to create a 256 bits key as defined in [ecdh-1pu](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-03#section-2.2)      |
```

how does this look?

Reply:
I'd feel more comfortable with a followup pr, submitted by somebody smarter about this stuff than I.

Reply:
Well, I got excited and included the table you included.